src/theme.rs

```diff
@@ -456,6 +456,81 @@ pub trait Theme {
         )
     }
 
+    /// Returns the progressbar start style for the [`indicatif::ProgressBar`].
+    fn format_progressbar_start(&self) -> String {
+        format!(
+            "{}\n{S_BAR}  {}",
+            "{spinner:.magenta}  {msg}", "[{elapsed_precise}] {bar:40.cyan/blue} ({pos}/{len})"
+        )
+    }
+
+    /// Returns the progressbar start style for the [`indicatif::ProgressBar`].
+    fn format_progressbar_stop(&self, msg: &str) -> String {
+        format!(
+            "{symbol}  {msg}\n{bar}",
+            symbol = self.state_symbol(&ThemeState::Submit),
+            bar = self.bar_color(&ThemeState::Submit).apply_to(S_BAR)
+        )
+    }
+
+    /// Returns the progressbar start style for the [`indicatif::ProgressBar`].
+    fn format_downloadbar_start(&self) -> String {
+        format!(
+            "{}\n{S_BAR}  {}",
+            "{spinner:.magenta}  {msg}",
+            "[{elapsed_precise}] [{bar:40.cyan/blue}] {bytes}/{total_bytes} ({eta})"
+        )
+    }
+
+    /// Returns the progressbar start style for the [`indicatif::ProgressBar`].
+    fn format_downloadbar_stop(&self, msg: &str) -> String {
+        format!(
+            "{symbol}  {msg}\n{bar}",
+            symbol = self.state_symbol(&ThemeState::Submit),
+            bar = self.bar_color(&ThemeState::Submit).apply_to(S_BAR)
+        )
+    }
+
+    /// Returns the progressbar with a final message in a specified state.
+    ///
+    /// It's not symmetric to [`Theme::format_downloadbar_start`] because of a workaround
+    /// for the [`indicatif::ProgressBar`] progressbar behavior which disrupts
+    /// the line after the stop message reproduced while terminal resizing
+    /// (see [`Spinner::stop`](fn@crate::Spinner::stop)).
+    fn format_downloadbar_with_state(
+        &self,
+        msg: &str,
+        state: &ThemeState,
+    ) -> std::io::Result<String> {
+        let term = Term::stderr();
+        term.move_cursor_up(1)?;
```

Comment:
This is not good, as `src/theme.rs` is de-coupled with any Terminal interfaces/abstractions. It must not be here.
You won't need it if you agree to implement a one-liner progress bar.

Reply:
I agree it's ugly -- but from wishing a "uniform UI experience" where progress bars are in alignment with each other (think the `indicatif:multi` example), I'm not sure how this should be handled while maintaining the **beautiful** look of `cliclack`. I'm admittedly only an hour or so deep in your codebase, so I'm hoping that you might have some bright ideas here? :smile:

Reply:
Moved this to `progressbar.rs`-- is it OK there?